### DISCUSSION TITLE
Incremental Static (Re)Generation

### DISCUSSION THREAD

**Discussion by pbuzdin** [ElMassimo/iles#48] [Ideas]
Generate and regenerate only pages and partials which should be updated.

**Additional context**
https://vercel.com/docs/concepts/next.js/incremental-static-regeneration

**Comment by ElMassimo** [comment]
> Incremental builds 

Do you mean incremental rendering?

Currently îles [deletes the server bundle](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/node/build/build.ts#L28) after build, but it would possible to keep it instead and provide a convenience script that can be used to render [dynamic routes](https://iles-docs.netlify.app/guide/routing#dynamic-routes) [that were not prerendered](https://docs.layer0.co/guides/isg#section_adding_isg_to_an_app_built_with_any_framework) in `getStaticPaths`.

As to integrating specifically with providers like Vercel, or Netlify, that's out of scope for the core of îles, and could be implemented as [modules](https://iles-docs.netlify.app/config/plugins).
